test/casttest.c

```diff
@@ -63,101 +63,54 @@ static unsigned char c_b[16] = {
     0x80, 0xAC, 0x05, 0xB8, 0xE8, 0x3D, 0x69, 0x6E
 };
 
-int main(int argc, char *argv[])
+static int cast_test(void)
 {
-# ifdef FULL_TEST
+    static char *hex = "0123456789ABCDEF";
     long l;
-    CAST_KEY key_b;
-# endif
-    int i, z, err = 0;
-    CAST_KEY key;
+    int i, z, testresult = 1;
+    CAST_KEY key, key_b;
+    unsigned char out_a[16], out_b[16];
 
     for (z = 0; z < 3; z++) {
         CAST_set_key(&key, k_len[z], k);
-
         CAST_ecb_encrypt(in, out, &key, CAST_ENCRYPT);
-        if (memcmp(out, &(c[z][0]), 8) != 0) {
-            printf("ecb cast error encrypting for keysize %d\n",
-                   k_len[z] * 8);
-            printf("got     :");
-            for (i = 0; i < 8; i++)
-                printf("%02X ", out[i]);
-            printf("\n");
-            printf("expected:");
-            for (i = 0; i < 8; i++)
-                printf("%02X ", c[z][i]);
-            err = 20;
-            printf("\n");
-        }
+        if (!TEST_mem_eq(out, sizeof(c[z]), c[z], sizeof(c[z])))
```

Comment:
That, or `Z`, but quite frankly, `k_len[Z]` gives a more intuitive indication of what was done.  Either way, it shows which iteration of the loop went wrong.